gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -318,9 +326,9 @@ public static List<Service> parseServices(
       apiDefinedRpcs.addAll(
           service.methods().stream().map(m -> m.name()).collect(Collectors.toSet()));
     }
-    // Mix-in APIs only if the protos are present and they're defined in the service.yaml file.
+    // Mix-in APIs if they're defined in the service.yaml file.
     Set<Service> outputMixinServiceSet = new HashSet<>();
-    if (servicesContainBlocklistedApi && !mixedInApis.isEmpty()) {
+    if (servicesContainBlocklistedApi) {
```

Comment:
`!mixedInApis.isEmpty()` check is still required, see my [comments](https://github.com/googleapis/sdk-platform-java/pull/2194/files#r1376566678) in the tests.

Reply:
done.